rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -48,38 +48,56 @@ get_node_names(
 
   DDS::DomainParticipant * participant = static_cast<ConnextNodeInfo *>(node->data)->participant;
   DDS::InstanceHandleSeq handles;
+
   if (participant->get_discovered_participants(handles) != DDS::RETCODE_OK) {
     RMW_SET_ERROR_MSG("unable to fetch discovered participants.");
     return RMW_RET_ERROR;
   }
+
+  DDS::DomainParticipantQos participant_qos;
+  DDS::ReturnCode_t status = participant->get_qos(participant_qos);
+
+  if (status != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("failed to get default participant qos");
+    return RMW_RET_ERROR;
+  }
+
+  rmw_ret_t final_ret = RMW_RET_OK;
+
   auto length = handles.length() + 1;  // add yourself
+
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
-  rcutils_ret_t rcutils_ret = rcutils_string_array_init(node_names, length, &allocator);
+
+  // Pre-allocate temporary buffer for all node names & namespaces
+  // Nodes that are not created by ROS 2 could provide empty names
+  // Such names should not be returned
+  rcutils_string_array_t tmp_names_list = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t tmp_namespaces_list = rcutils_get_zero_initialized_string_array();
+
+  int named_nodes_num = 1;
```

Comment:
why this starts from 1?

Reply:
The first element is set to the current node name at L96. I guess for clarity this could have started at zero and then been incremented after L96.

Reply:
Yes, didn't realize.
The logic is a bit confusing.